src/amqp.rs

```diff
@@ -34,6 +34,29 @@ impl Default for AmqpConfig {
     }
 }
 
+/// Polls the connection until a connection can be made.
+/// Gives up after 5 attempts.
+#[cfg(test)]
+pub(crate) async fn await_connection(connection: &AmqpConfig) {
+    let mut pause = tokio::time::Duration::from_secs(1);
+    let mut attempts = 0;
+
+    loop {
+        let connection = connection.clone();
+        if connection.connect().await.is_ok() {
+            return;
+        }
+        attempts += 1;
+
+        if attempts == 5 {
+            return;
+        }
+
+        tokio::time::sleep(pause).await;
+        pause *= 2;
+    }
+}
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏Not an issue just an observation that this is a long time. Were these durations/intervals experimented with? How often does it take more than one attempt?

Reply:
No, there wasn't any experimentation. I imagine it would be largely machine dependant, but I can try reducing it and see if it still works.

Reply:
Ok. I've reduced the pause significantly.